Reportback fixes part 3a

weerd opened the pull request:
Resolves #3808

Wrapping up most of the markup and logic when campaign loads with/or without having user reportbacks in Prove It section.

If there are no reportbacks from the user, then show the large button (which still needs further styling):
![screen shot 2015-01-27 at 12 27 49 pm](https://cloud.githubusercontent.com/assets/105849/5923388/2254f432-a621-11e4-97c7-dd339cf7b0ce.png)

Otherwise, if there are prior reportback submissions, then show the last submission prominently, allow user to "Add another photo" (file upload dialogue), and show prior submissions:
![screen shot 2015-01-27 at 12 27 11 pm](https://cloud.githubusercontent.com/assets/105849/5923419/60ff1d5c-a621-11e4-9e53-02ac42a6faab.png)

@aaronschachter @sbsmith86 
CC: @barryclark @DFurnes

Commit:
Update to wrap form elments in a container.

Commit:
Update to add wrapping container around components.

Commit:
Wrapping up markup and layout for form functionality and some basic styles.

weerd commented:
FYI: More styling needs to happen, so kind of ignore some of the funkyness and lack of whitespace etc, for now ;) huggles!

aaronschachter commented on lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc:
```diff
@@ -53,26 +54,76 @@ function dosomething_reportback_form($form, &$form_state, $entity = NULL) {
   }
-  // Else, it's update form.
+
+  // Else, it's update an form.
```
An update form

weerd replied:
Lol. I had it fixed correctly before and then Vagrant crapped out on me and I had to redo some code. Put the 'an' in the wrong place, haha.

aaronschachter commented on lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc:
```diff
@@ -378,4 +422,4 @@ function dosomething_reportback_flag_confirm_submit($form, &$form_state) {
   // Log the flag operation.
   $reportback->insertLog($op);
-}
+}
```
:cactus:

weerd replied:
Not sure what this was. My editor being funky?

aaronschachter replied:
Looks like you removed the blank line, https://www.drupal.org/node/318#indenting

sbsmith86 commented on lib/modules/dosomething/dosomething_reportback/theme/reportback-latest-submission.tpl.php:
```diff
@@ -0,0 +1 @@
+<?php print $image; ?>
```
new line?

weerd replied:
This is just for the template in the module, that doesn't get used cause it's overridden in the theme ;)

aaronschachter replied:
Same here

aaronschachter commented:
Looks ok, assuming this has been tested and works when the relevant theme/module settings are turned off and this is safe to deploy without turning on.

weerd commented:
Oh, good call. Lemme check on that.

Commit:
Grammar fix.

Commit:
Adding proper fixes to account for feature flags for Production.

weerd commented:
@aaronschachter Ok! Updated to account for feature flags. I had to use a large `if/else` conditional for part of the form to hide the new approach from Production. I added comments and an esaily findable `@TODO_RBv2` comment to know where things need to be edited and removed post-launch!

aaronschachter commented:
Nice PHP :+1:  Feel free to fix line endings in subsequent PR to get this out.
